ReactAndroid/src/main/java/com/facebook/react/ReactInstanceManagerImpl.java

```diff
@@ -466,6 +466,22 @@ private void invokeDefaultOnBackPressed() {
     }
   }
 
+  @Override
+  public void onNewIntent(Intent intent) {
+    if (mCurrentReactContext == null) {
+      // Swallow the new intent?
+      FLog.w(ReactConstants.TAG, "Instance detached from instance manager");
+    } else {
+      String action = intent.getAction();
+
+      if (Intent.ACTION_VIEW.equals(action)) {
+        DeviceEventManagerModule deviceEventManagerModule =
```

Comment:
We should check for if uri is null, and shouldn't send the event in that case. Can you copy the code from here instead? https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/intent/IntentModule.java#L51